library/shelve.po

```diff
@@ -6,199 +6,288 @@ msgstr ""
+"but, if many entries are accessed, it can consume vast amounts of memory for the "
+"cache, and it can make the close operation very slow since all accessed entries "
+"are written back (there is no way to determine which accessed entries are "
+"mutable, nor which ones were actually mutated)."
 msgstr ""
+"À cause de la sémantique Python, un *shelf* ne peut pas savoir lorsqu'une entrée "
+"modifiable de dictionnaire persistant est modifiée. Par défaut les objets "
+"modifiés sont écris *seulement* lorsqu'ils sont assignés à une \"shelf\" (voir :"
+"ref:`shelve-example`). Si le paramètre optionnel *writeback* est mis à ``True``, "
+"toutes les entrées déjà accédées sont aussi mises en cache en mémoire, et ré-"
+"écrite sur :meth:`~Shelf.sync` et :meth:`~Shelf.close` ; cela peut faciliter la "
+"modification des entrées modifiables dans le dictionnaire persistant, mais, si "
+"beaucoup d'entrées sont déjà accédées, cela peut consommer beaucoup de mémoire "
+"cache, et cela peut rendre l'opération de fermeture très lente puisque toutes "
+"les entrées déjà accédées sont ré-écrites (il n'y a aucun moyen de savoir "
+"quelles entrées déjà accédées sont mutables, ni lesquelles ont été vraiment "
+"mutées)."
 
 #: ../Doc/library/shelve.rst:45
 msgid ""
-"Do not rely on the shelf being closed automatically; always call :meth:"
-"`~Shelf.close` explicitly when you don't need it any more, or use :func:"
-"`shelve.open` as a context manager::"
+"Do not rely on the shelf being closed automatically; always call :meth:`~Shelf."
+"close` explicitly when you don't need it any more, or use :func:`shelve.open` as "
+"a context manager::"
 msgstr ""
+"Ne pas se fier au *shelf* fermé automatiquement ; toujours appeler :meth:`~Shelf."
+"close` explicitement quand vous n'en avez plus besoin, ou utiliser :func:`shelve."
+"open` comme un gestionnaire de contexte ::"
 
 #: ../Doc/library/shelve.rst:54
 msgid ""
-"Because the :mod:`shelve` module is backed by :mod:`pickle`, it is insecure "
-"to load a shelf from an untrusted source.  Like with pickle, loading a shelf "
-"can execute arbitrary code."
+"Because the :mod:`shelve` module is backed by :mod:`pickle`, it is insecure to "
+"load a shelf from an untrusted source.  Like with pickle, loading a shelf can "
+"execute arbitrary code."
 msgstr ""
+"Puisque le module :mod:`shelve` est soutenu par :mod:`pickle`, ce n'est pas "
+"sécurisé de charger un *shelf* depuis une source non fiable. Comme avec \"pickle"
+"\", charger un *shelf* peut exécuter du code arbitraire."
 
 #: ../Doc/library/shelve.rst:58
 msgid ""
 "Shelf objects support all methods supported by dictionaries.  This eases the "
-"transition from dictionary based scripts to those requiring persistent "
-"storage."
+"transition from dictionary based scripts to those requiring persistent storage."
 msgstr ""
+"Les objets *shelf* supportent toutes les méthodes supportées sur les "
```

Comment:
Dans la documentation, nous traduisons généralement "to support" par "gérer". Ici :
"Les objets *shelf* gèrent toutes les méthodes des dictionnaires."